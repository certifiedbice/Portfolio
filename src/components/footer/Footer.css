#site-footer{
	position: fixed;
	display: block;
	bottom: 0;
	left: 0;
	z-index: 98;
	width: 100%;
	height: 45px;
	background-color: #fff;
	padding: 5px;
	text-align: center;
	font-family: Verdana, Geneva, Tahoma, sans-serif;
	font-size: 12px;
	color: #000;
	box-shadow: 0 -8px 8px #000;
}
#footer-links-container{
	margin-top: 5px;
}
.footer-social-icons-image{
	height: 25px;
	width: 25px;
}