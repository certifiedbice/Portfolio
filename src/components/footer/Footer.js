import React,{Component} from 'react';
import DataHelper from '../../dataHelper';
import './Footer.css';
export default class Header extends Component{
	render(){
		const socialInfo=DataHelper.social;
		return(
			<footer id='site-footer'>
				Bice - 2020
				<div id="footer-links-container">
					{
						socialInfo.map(item=>{
							if(item.link!==null){
								return (
									<a key={item.title} href={item.link} className='footer-social-icon-link' target='_blank' rel='noopener noreferrer'>
										<img className='footer-social-icons-image' src={require(`../../images/social-icons/${item.icon}`)} alt={item.title}/>
									</a>
								);
							}
							else return null;
						})
					}
				</div>
			</footer>
		);
	}
}