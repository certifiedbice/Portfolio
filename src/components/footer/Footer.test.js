import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import Footer from './Footer';

it('renders without crashing',()=>{
    const footer=document.createElement('footer');
    ReactDOM.render(<BrowserRouter><Footer/></BrowserRouter>,footer);
});

it('renders correctly',()=>{
    const tree=renderer
        .create(<BrowserRouter><Footer/></BrowserRouter>)
        .toJSON();
        expect(tree).toMatchSnapshot();
});