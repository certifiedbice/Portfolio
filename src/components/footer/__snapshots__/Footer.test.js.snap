// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<footer
  id="site-footer"
>
  Bice - 2020
  <div
    id="footer-links-container"
  >
    <a
      className="footer-social-icon-link"
      href="https://www.facebook.com/certifiedbice"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Facebook"
        className="footer-social-icons-image"
        src="facebook.png"
      />
    </a>
    <a
      className="footer-social-icon-link"
      href="https://github.com/certifiedbice"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="GitHub"
        className="footer-social-icons-image"
        src="github.png"
      />
    </a>
    <a
      className="footer-social-icon-link"
      href="https://www.instagram.com/certifiedbice"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Instagram"
        className="footer-social-icons-image"
        src="instagram.png"
      />
    </a>
    <a
      className="footer-social-icon-link"
      href="https://www.hackerrank.com/certifiedbice"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="HackerRank"
        className="footer-social-icons-image"
        src="hackerrank.png"
      />
    </a>
    <a
      className="footer-social-icon-link"
      href="https://www.linkedin.com/in/certifiedbice"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="LinkedIn"
        className="footer-social-icons-image"
        src="linkedin.png"
      />
    </a>
    <a
      className="footer-social-icon-link"
      href="https://twitter.com/certifiedbice"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Twitter"
        className="footer-social-icons-image"
        src="twitter.png"
      />
    </a>
    <a
      className="footer-social-icon-link"
      href="https://www.youtube.com/channel/UCEITFHDFDtvTIKFbUhG_E9w"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="YouTube"
        className="footer-social-icons-image"
        src="youtube.png"
      />
    </a>
  </div>
</footer>
`;
